test/compress/dead-code.js

```diff
@@ -855,3 +855,28 @@ issue_2666: {
     }
     expect_stdout: "object"
 }
+
+issue_2692: {
+    options = {
+        dead_code: true,
+    }
+    input: {
+        function f(a) {
+            return a = g;
+            function g() {
+                return a;
+            }
+        }
+        console.log(typeof f()());
```

Comment:
This test case produces the correct result with `v3.3.3`.

Reply:
Yes, this test case demonstrates the issue only when `reduce_vars` is disabled.

Reply:
```sh
$ git checkout v3.3.3
$ cat test.js
function f(a) {
    return a = g;
    function g() {
        return a;
    }
}
console.log(typeof f()());

$ cat test.js | node
function

$ uglifyjs test.js -bc reduce_vars=0
function f(a) {
    return g;
    function g() {
        return a;
    }
}

console.log(typeof f()());

$ uglifyjs test.js -bc reduce_vars=0 | node
undefined
```

Reply:
That fools me every time. Could you explicitly add `reduce_vars: false,` in the options to reinforce that?

Reply:
Done in https://github.com/mishoo/UglifyJS2/pull/2693/commits/74892b7a79d59b3ae33bed5a37bfb2fd263c5c3f